sympy/physics/control/lti.py

```diff
@@ -1079,78 +1328,51 @@ def doit(self, **kwargs):
         TransferFunction((2 - s**3)*(-p + s) + (-a*p**2 - b*s)*(s**4 + 5*s + 6), (-p + s)*(s**4 + 5*s + 6), s)
 
         """
-        res = None
-        for arg in self.args:
-            arg = arg.doit()
-            if res is None:
-                res = arg
-            else:
-                num_ = res.num * arg.den + res.den * arg.num
-                den_ = res.den * arg.den
-                res = TransferFunction(num_, den_, self.var)
-        return res
+
+        _arg = (arg.doit().to_expr() for arg in self.args)
+        res = Add(*_arg).as_numer_denom()
+        return TransferFunction(*res, self.var)
```

Comment:
Can this lead to pole-zero cancellation?

Reply:
I tried a few cases where there was a possibility of pole-zero cancelation, but that didn't happen.